Python/Iniciante/1183.py

```diff
@@ -0,0 +1,33 @@
+# Problema 1183 - Beecrowd - Iniciante - Nível 4
+
+# Leitura de um caractere maiúsculo que especifica a operação
+op = input()
+
+# Leitura e Construção da matriz M[12][12]
+M = [[float(input()) for x in range(12)] for y in range(12)]
+
+# Soma total
+soma = 0
+
+# Variável auxiliar para limitar o acesso a somente colunas que
+# possuem elementos acima da diagonal principal
+aux = 1
+
+# Cálculo da soma dos elementos acima da diagonal principal
+for i in range(0, 12):
+    for j in range(aux, 12):
+        if j > i:
+            soma += M[i][j]
+    aux += 1
```

Comment:
O exercício está bom, mas podemos utilizar um macete do python para evitar 2 "fors". Que é:

Quando você usa `lista[numA:numB:numC]`, você está pegando partes da sua lista com a seguinte propriedade:
- O python vai começar a sua "sublista" do índice "numA", omitir o "numA" faz com que ele pegue a primeira posição do vetor.
- O python termina a "sublista" no índice "numB", omitir o "numB" faz com que ele pegue a última posição do vetor.
- O python vai "andar" os índices entre "numC" vezes, omitir o "numC" faz com que ele ande de 1 em 1 posição no vetor.

Por exemplo se você tiver uma `lista = [0,1,2,3,4,5]` e fizer:
- `print(lista[:3:2])` == `[0,2]`, pois ele vai começar do índice 0 , ele vai até o índice 3 e vai "andar" de 2 em 2.
- `print(lista[1::3])` == `[1,4]`, pois ele vai começar do índice 1, ele vai até o final e vai andar de 3 em 3.
- `print(lista[::-1])` == `[5,4,3,2,1,0]`, pois ele vai começar do índice 0, vai até o final e vai andar de -1 em -1. _(Esse é um jeito muito simples de inverter vetores em python)_

```suggestion
# Cálculo da soma dos elementos acima da diagonal principal
for i in range(11):
    
    # A função "sum()" soma todos os valores de um objeto iterável, então
    # estamos passando a linha "i", começando do valor "aux" até o seu fim.
    soma += sum(M[i][aux:])
    aux += 1
```

Reply:
Entendi, Vitor! Ficou bem mais apresentável. Não tinha pensado dessa forma utilizando a função ```sum()``` com manipulação de "sublistas", ficou muito bom. Vi agora também que o "for" anterior que criei, responsável por percorrer as linhas da matriz estava com uma iteração a mais do que o necessário. 
Muito obrigado pela explicação e pela correção! Vou fazer o commit das alterações, ok?

Reply:
Beleza! Para fazer o commit nas alterações, você precisa clicar no botãozinho acima do seu comentário com o "Commit suggestion". Que ele vai aplicar automaticamente no seu código.

Reply:
Feito! Muito Obrigado!